scrapi/base/__init__.py

```diff
@@ -241,14 +242,8 @@ def normalize(self, raw_doc):
                 'ns0:header/ns0:setSpec/node()',
                 namespaces=self.NAMESPACES
             )
-            approved = False
-            for item in set_spec:
-                item_mod = item.replace('publication:', '')
-                if item_mod in self.approved_sets:
-                    approved = True
-                else:
-                    logger.info('Series {} not in approved list'.format(item))
-            if not approved:
+            if not set([x.replace('publication:', '') for x in set_spec]).intersection(self.approved_sets):
```

Comment:
There is a lot going on here. It might be nice to add a comment explaining what is happening in that list/set comprehension. Also, is there any reason you are doing a list comprehension and converting it to a set, rather than just a set comprehension? https://docs.python.org/2/tutorial/datastructures.html#sets